### PR TITLE
[Improvement-16746][seatunnel] pass user defined task parameter to seatunnel

### LINKED ISSUE #16746: [Bug] [SeaTunnel-Task] can't append variables to seatunnel shell command
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar issues.

### What happened

In my seatunnel config, there has some variables need replace on run time by parameter.

image

I defined the parameter in custom parameter setting.

image

But this parameter not append to the shell command.

```
SeaTunnel task command: /Users/jarvis/Downloads/apache-seatunnel-2.3.8/bin/seatunnel.sh --config /tmp/dolphinscheduler/exec/process/default/15454113618464/15454122106144_1/1/1/seatunnel_1_1.conf --deploy-mode local
```

I check the code, it can get the user defined parameter from localParameters

image

So we need check if localParameters not empty, then append `-i key='value'` parameter to the shell command.

### What you expected to happen

fix seatunnel task submit issue

### How to reproduce

run a seatunnel task

### Anything else

_No response_

### Version

dev

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by liunaijie**
## Purpose of the pull request
close #16746

pass the user defined parameter in local parameter, global parameter to seatunnel shell command.

## Verify this pull request

This pull request is code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

(or)

## Pull Request Notice
[Pull Request Notice](https://github.com/apache/dolphinscheduler/blob/dev/docs/docs/en/contribute/join/pull-request.md)

If your pull request contain incompatible change, you should also add it to `docs/docs/en/guide/upgrede/incompatible.md`

### CODE REVIEW

**Comment by liunaijie** [comment]
will add test cast when this pr https://github.com/apache/dolphinscheduler/pull/16741 merged

File: dolphinscheduler-task-plugin/dolphinscheduler-task-seatunnel/src/main/java/org/apache/dolphinscheduler/plugin/task/seatunnel/SeatunnelTask.java [hunk 1]
```diff
@@ -169,9 +170,33 @@ protected List<String> buildOptions() throws Exception {
         String filePath = buildConfigFilePath();
         createConfigFileIfNotExists(scriptContent, filePath);
         args.add(filePath);
+        args.addAll(generateTaskParameters());
         return args;
     }
 
+    private List<String> generateTaskParameters() {
```
Comment (SbloodyS) [inline dolphinscheduler-task-plugin/dolphinscheduler-task-seatunnel/src/main/java/org/apache/dolphinscheduler/plugin/task/seatunnel/SeatunnelTask.java:178]: Please add these feature to the docs.
https://github.com/apache/dolphinscheduler/blob/dev/docs/docs/zh/guide/task/seatunnel.md
https://github.com/apache/dolphinscheduler/blob/dev/docs/docs/en/guide/task/seatunnel.md

**Review by SbloodyS — DISMISSED** [review]
LGTM

File: dolphinscheduler-task-plugin/dolphinscheduler-task-seatunnel/src/main/java/org/apache/dolphinscheduler/plugin/task/seatunnel/SeatunnelTask.java [hunk 2]
```diff
@@ -169,9 +170,33 @@ protected List<String> buildOptions() throws Exception {
...
+        Map<String, String> variables = new HashMap<>();
+        Map<String, Property> paramsMap = taskExecutionContext.getPrepareParamsMap();
+        List<Property> propertyList = JSONUtils.toList(taskExecutionContext.getGlobalParams(), Property.class);
+        if (propertyList != null && !propertyList.isEmpty()) {
+            for (Property property : propertyList) {
+                variables.put(property.getProp(), paramsMap.get(property.getProp()).getValue());
+            }
+        }
```
Comment (ruanwenjun) [inline dolphinscheduler-task-plugin/dolphinscheduler-task-seatunnel/src/main/java/org/apache/dolphinscheduler/plugin/task/seatunnel/SeatunnelTask.java:186]: We need to filter the out type here.

**Review by ruanwenjun — CHANGES_REQUESTED** [review]

**Review by ruanwenjun — APPROVED** [review]
LGTM

**Review by SbloodyS — APPROVED** [review]
+1

File: docs/docs/en/guide/task/seatunnel.md [hunk 3]
```diff
@@ -26,17 +26,16 @@ Click [here](https://seatunnel.apache.org/) for more information about `Apache S
 - SEATUNNEL_ENGINE
 - Deployment mode: specify the deployment mode, `cluster` `local`
 
-          > Click [here](https://seatunnel.apache.org/docs/2.3.3/command/usage) for more information on the usage of 
-
-`Apache SeaTunnel command`
+  > Click [here](https://seatunnel.apache.org/docs/2.3.3/command/usage) for more information on the usage of Apache SeaTunnel command`
```
Comment (Hisoka-X) [inline docs/docs/en/guide/task/seatunnel.md:29]: Hi @liunaijie , how about update the doc version to 2.3.8?

Comment (liunaijie) [reply docs/docs/en/guide/task/seatunnel.md:29]: We can remove the version path, update to 
https://seatunnel.apache.org/docs/command/usage/

This link will always avaliable.

Comment (Hisoka-X) [reply docs/docs/en/guide/task/seatunnel.md:29]: Yes, this way would be better.

Comment (liunaijie) [reply docs/docs/en/guide/task/seatunnel.md:29]: I will update it

**Pull request merged**